grants/teonite/m1.md

```diff
@@ -0,0 +1,80 @@
+
+This is an RnD (Reasearch & Development) milestone and the goal is to tackle this critical issue and recreate or reverse the pre-processing to compare with the binary created by compiling the source code.
+Our analysis shows (please see the detailed analysis chapter below: WASM binary processing analysis), that we should be able to recreate the binary optimization process performed on the blockchain node while executing the finalized block but that needs actual RnD work to confirm it (or state otherwise) - thus this milestone.
+
+There also was uncertainty about the Rust compiler itself producing deterministic builds - but from our research code translation methods are now based on pseudocode for a very long time - in other words, the compiler works deterministically by translating to its higher-level pseudocode, optimizing it and then switching to real asm.
+
+
+### Acceptance Criteria
+
+Deliverables:
+
+1. Recreate the binary optimization process performed on the blockchain node while executing the finalized block with a Rust crate and a simple command line tool that uses the same mechanisms as the casper-node.
+
+2. Create tests that will either prove that it’s possible to produce the same WASM build as the node does or will fail and state there is no possibility to do so without chaining the node/core CSPR services.
+
+If the reproduction process succeeds - next milestones are valid, if the reproduction process and tests fail next milestones will not be realized.
+
+
+
+WASM binary processing analysis
+Below is a general description of how the smart contract binary is processed within the system before being stored on the blockchain. The analysis is based on the latest releases of relevant components:
+casper-node v1.5.0-rc.1 (a50b85a)
+casper-client-rs v1.6.0 (63a311a)
+casper-wasm-utils v1.0.0 (035ba8c)
+WASM/Deploy processing flow
+developer compiles smart contract source code into a .wasm file
+developer sends the binary to a node as part of a Deploy by using the casper-client CLI
+https://docs.casper.network/resources/quick-start/
+the client attaches the binary as-is and sends the Deploy to a JSON-RPC endpoint on the specified node
+https://github.com/casper-ecosystem/casper-client-rs/blob/63a311afbd3e29d3424b4eb2f620468603a17510/lib/parsing.rs#L507
+https://github.com/casper-ecosystem/casper-client-rs/blob/63a311afbd3e29d3424b4eb2f620468603a17510/lib/parsing.rs#L606
+request is received and RpcServerAnnouncement::DeployReceived event is added to API event queue; the event is actually immediately converted to Event::Accept type associated with MainReactor
+https://github.com/casper-network/casper-node/blob/a50b85a910a4c49b60e5cb2402de336d85aff6c6/node/src/components/rpc_server.rs#L275
+https://github.com/casper-network/casper-node/blob/a50b85a910a4c49b60e5cb2402de336d85aff6c6/node/src/components/deploy_acceptor/event.rs#L103
+DeplyAcceptor receives a new Deploy and verifies it complies with chain configuration and is not expired
```

Comment:
One more typo nitpicking, it's mentioned as **DeplyAcceptor**, instead of **DeployAcceptor**. This typo appears in all the occurrences of **DeplyAcceptor**

Reply:
I'm not sure if this is a typo or if they really name it like this. Let's leave it.